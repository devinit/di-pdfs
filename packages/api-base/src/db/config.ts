import {TConfig} from 'pg-promise';

// get the connection details from Allan or Donata

export const dwConfig: TConfig = {
    user: process.env.DB_USER,
    database: process.env.DB_DATABASE,
    host: process.env.DB_HOST,
    port: Number(process.env.DB_PORT),
    password: process.env.DB_PASSWORD
};
