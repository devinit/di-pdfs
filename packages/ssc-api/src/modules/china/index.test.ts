import * as prettyFormat from 'pretty-format';
import {China} from '.';
import db from '@di-pdfs/api-base/lib/db';

describe('China', () => {
    const china = new China(db);
    it('getting dvptCooperation', async () => {
       const data = await china.dvptCooperation();
       expect(prettyFormat(data)).toMatchSnapshot();
    }, 100000);
    it('getting multiCooperation', async () => {
        const data = await china.odaLikeFlowsByRegion();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    it('getting tch top recipients', async () => {
        const data = await china.odaLikeFlowsBySector();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    it('getting tch top recipients', async () => {
        const data = await china.odaRecipients();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    it('getting tch top recipients', async () => {
        const data = await china.overallMultiExpenditure();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    afterAll(() => {
       db.$config.pgp.end();
    });
});
