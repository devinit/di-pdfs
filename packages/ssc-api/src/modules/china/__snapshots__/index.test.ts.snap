// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`China getting odaLikeFlowsByRegion 1`] = `
"Array [
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 65.9,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 19,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"region\\": \\"Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 7,
  },
  Object {
    \\"color\\": \\"#21BA45\\",
    \\"region\\": \\"Americas\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 6,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"region\\": \\"Oceania\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 2.1,
  },
]"
`;

exports[`China getting odaLikeFlowsBySector 1`] = `
"Array [
  Object {
    \\"color\\": \\"#0071b1\\",
    \\"sector\\": \\"Transport and Storage\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 5695.6,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#bc2629\\",
    \\"sector\\": \\"Energy Generation and Supply\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1441.9,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#21BA45\\",
    \\"sector\\": \\"Industry, Mining, Construction\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1355.8,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"sector\\": \\"Water Supply and Sanitation\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 919.8,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#DC73FF\\",
    \\"sector\\": \\"Communications\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 306.8,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"sector\\": \\"Trade and Tourism\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"sector\\": \\"Government and Civil Society\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 445.7,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"sector\\": \\"All others\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1777.7,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#0071b1\\",
    \\"sector\\": \\"Transport and Storage\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3319.9,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#bc2629\\",
    \\"sector\\": \\"Energy Generation and Supply\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 568.5,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#21BA45\\",
    \\"sector\\": \\"Industry, Mining, Construction\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 187.1,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"sector\\": \\"Water Supply and Sanitation\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 560.3,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#DC73FF\\",
    \\"sector\\": \\"Communications\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 696,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"sector\\": \\"Trade and Tourism\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 900.3,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"sector\\": \\"Government and Civil Society\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 276.6,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"sector\\": \\"All others\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3046,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#0071b1\\",
    \\"sector\\": \\"Transport and Storage\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 4008.4,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#bc2629\\",
    \\"sector\\": \\"Energy Generation and Supply\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1404,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#21BA45\\",
    \\"sector\\": \\"Industry, Mining, Construction\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 439,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"sector\\": \\"Water Supply and Sanitation\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#DC73FF\\",
    \\"sector\\": \\"Communications\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 367,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"sector\\": \\"Trade and Tourism\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 31.7,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"sector\\": \\"Government and Civil Society\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 28.2,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"sector\\": \\"All others\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 579.8,
    \\"year\\": 2014,
  },
]"
`;

exports[`China getting odaRecipients 1`] = `
"Array [
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Nigeria\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1628.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Zimbabwe\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1455,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Niger\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1223.3,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Cameroon\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1137.8,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Sri Lanka\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1023.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Mozambique\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 952.9,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"recipient\\": \\"Montenegro\\",
    \\"region\\": \\"Central and Eastern Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 911.5,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Ethiopia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 884.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Sudan\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 700.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Africa, regional\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 539.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Uganda\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 512.9,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Cambodia\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 496.9,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Mongolia\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 496.7,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Pakistan\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 475.6,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Indonesia\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 465.7,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Zambia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 415.5,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Costa Rica\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 401.6,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Ghana\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 396.2,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Uzbekistan\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 364.3,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Jamaica\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 329.9,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Nepal\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 327.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Kenya\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 324.7,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Bangladesh\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 317.8,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"recipient\\": \\"Serbia\\",
    \\"region\\": \\"Central and Eastern Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 310.3,
  },
  Object {
    \\"color\\": \\"#FF8EDF\\",
    \\"recipient\\": \\"Fiji\\",
    \\"region\\": \\"The Pacific\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 303.2,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Mauritania\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 297.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Namibia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 240,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Rwanda\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 215.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Mali\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 211.8,
  },
  Object {
    \\"color\\": \\"#FF8EDF\\",
    \\"recipient\\": \\"Vanuatu\\",
    \\"region\\": \\"The Pacific\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 180.1,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Guyana\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 156.2,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Gabon\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 133.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Angola\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 117.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Sierra Leone\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 115.1,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"recipient\\": \\"Belarus\\",
    \\"region\\": \\"Central and Eastern Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 102.3,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Tajikistan\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 73.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Benin\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 67.8,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"South Sudan\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 58.7,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Senegal\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 53.2,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"recipient\\": \\"Ukraine\\",
    \\"region\\": \\"Central and Eastern Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 52,
  },
  Object {
    \\"color\\": \\"#FF8EDF\\",
    \\"recipient\\": \\"Samoa\\",
    \\"region\\": \\"The Pacific\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 50.3,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Afghanistan\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 33.5,
  },
  Object {
    \\"color\\": \\"#FF8EDF\\",
    \\"recipient\\": \\"Tonga\\",
    \\"region\\": \\"The Pacific\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 33.2,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Malawi\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 32.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Tunisia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 31.9,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Grenada\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 27.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Djibouti\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 26.6,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Burundi\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 25.3,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Egypt\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 24.4,
  },
  Object {
    \\"color\\": \\"#FF8EDF\\",
    \\"recipient\\": \\"Papua New Guinea\\",
    \\"region\\": \\"The Pacific\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 23.4,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Armenia\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 21.5,
  },
  Object {
    \\"color\\": \\"#E03997\\",
    \\"recipient\\": \\"Yemen\\",
    \\"region\\": \\"Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 21.1,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Myanmar\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 20.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Mauritius\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 19.3,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Equatorial Guinea\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 18.9,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Togo\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 18.7,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Dominica\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 18.7,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Lesotho\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 18.6,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Chad\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 16.1,
  },
  Object {
    \\"color\\": \\"#E03997\\",
    \\"recipient\\": \\"Jordan\\",
    \\"region\\": \\"Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 16.1,
  },
  Object {
    \\"color\\": \\"#E03997\\",
    \\"recipient\\": \\"Middle East, regional\\",
    \\"region\\": \\"Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 16,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Guinea\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 12,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Liberia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 9.9,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Maldives\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 4.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Morocco\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3.3,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Philippines\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1.6,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Guinea-Bissau\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1.6,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Comoros\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1.1,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Southeast Asia, regional\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Uruguay\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Cuba\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.4,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Thailand\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.1,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Peru\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.1,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Somalia\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Malaysia\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Botswana\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Viet Nam\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Colombia\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": \\"0\\",
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Antigua and Barbuda\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 6,
  },
  Object {
    \\"color\\": \\"#54C8FF\\",
    \\"recipient\\": \\"Bolivia (Plurinational State of)\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 761.5,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Cabo Verde\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 66.7,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Congo (Democratic Republic of the)\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 5,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Congo\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 887,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Côte d'Ivoire\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3508.2,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Korea (Democratic People's Republic of)\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 19.1,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Kyrgyzstan\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1109.9,
  },
  Object {
    \\"color\\": undefined,
    \\"recipient\\": \\"Lao People's Democratic Republic\\",
    \\"region\\": \\"Asia\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 89,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"recipient\\": \\"Macedonia (the former Yugoslav Republic of)\\",
    \\"region\\": \\"Central and Eastern Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 600.3,
  },
  Object {
    \\"color\\": \\"#E03997\\",
    \\"recipient\\": \\"Syrian Arab Republic\\",
    \\"region\\": \\"Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 5.4,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"recipient\\": \\"Tanzania, United Republic of\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 2280.7,
  },
]"
`;

exports[`China getting overallMultiExpenditure 1`] = `
"Array [
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.2,
    \\"year\\": 2006,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.2,
    \\"year\\": 2007,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.3,
    \\"year\\": 2008,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.3,
    \\"year\\": 2009,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.4,
    \\"year\\": 2010,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.3,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.4,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.7,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.4,
    \\"year\\": 2014,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"id\\": \\"Multilateral organisations\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.6,
    \\"year\\": 2015,
  },
]"
`;
